src/topp/MaRaClusterAdapter.cpp

```diff
@@ -0,0 +1,514 @@
+          }
+          all_peptide_ids.insert(all_peptide_ids.end(), peptide_ids.begin(), peptide_ids.end());
+          all_protein_ids.insert(all_protein_ids.end(), protein_ids.begin(), protein_ids.end());
+        }
+
+      }
+      else
+      {
+        for (Map<MaRaClusterResult,Int>::iterator sid = specid_to_clusterid_map.begin(); sid != specid_to_clusterid_map.end(); ++sid) {
+          Int scan_nr = sid->first.scan_nr;
+          Int file_id = sid->first.file_idx;
+          PeptideIdentification pid;
+          PeptideHit pih;
+          pid.insertHit(pih);
+          pid.setMetaValue("spectrum_reference", "scan=" + String(scan_nr));
+          pid.setMetaValue("cluster_id", sid->second);
+          pid.setMetaValue("file_origin", in_list[file_id]);
+          all_peptide_ids.push_back(pid);
+        }
+      }
+
+      if (all_protein_ids.size() == 0)
+      {
+        ProteinIdentification protid;
+        all_protein_ids.push_back(protid);
+      }
+
+      all_protein_ids.back().setMetaValue("maracluster", "MaRaClusterAdapter");
+      ProteinIdentification::SearchParameters search_parameters = all_protein_ids.back().getSearchParameters();
+
+      search_parameters.setMetaValue("MaRaCluster:pvalue-cutoff", pcut);
+
+      all_protein_ids.back().setSearchParameters(search_parameters);
+
+
+      writeDebug_("write idXMLFile", 1);
+      writeDebug_(out, 1);// As the maracluster output file is not needed anymore, the temporary directory is going to be deleted
+      IdXMLFile().store(out, all_protein_ids, all_peptide_ids);
+    }
+
+    if (!consensus_out.empty())
+    {
+      QStringList arguments_consensus;
+      // Check all set parameters and get them into arguments StringList
+      {
+        arguments_consensus << "consensus";
+        arguments_consensus << "-l" << consensus_output_file.toQString();
+        arguments_consensus << "-f" << temp_directory_body.toQString();
+        arguments_consensus << "-o" << consensus_out.toQString();
+        Int min_cluster_size = getIntOption_("min_cluster_size");
+        arguments_consensus << "-M" << String(min_cluster_size).toQString();
+
+        Int verbose_level = getIntOption_("verbose");
+        if (verbose_level != 2) arguments_consensus << "-v" << String(verbose_level).toQString();
+      }
+      writeLog_("Prepared maracluster-consensus command.");
+
+      //-------------------------------------------------------------
+      // run MaRaCluster
+      //-------------------------------------------------------------
+      // MaRaCluster execution with the executable and the arguments StringList
+      int status_consensus = QProcess::execute(maracluster_executable.toQString(), arguments_consensus); // does automatic escaping etc...
+      if (status_consensus != 0)
+      {
+        writeLog_("MaRaCluster problem. Aborting! Calling command was: '" + maracluster_executable + " " + arguments_consensus.join(" ").toStdString() + "'.");
```

Comment:
Can you please capture the QProcess output and log it here at a reasonable debug level or by default? See latest PRs by @oliveralka

Reply:
Does including the qp.exitCode() solve this?

Reply:
Not completely. I would like to see the QProcess::readStdOutput/Error